addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java

```diff
@@ -242,11 +243,38 @@ private void cacheBusterHttpMethod(String url, Method method, Cache cache)
                 if (msg.getResponseHeader().getStatusCode()
                         == base.getResponseHeader().getStatusCode()) {
                     String indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
+                    if (indicValue == null) {
+                        switch (method) {
+                            case GET:
+                                headers.setMethod(HttpRequestHeader.GET);
+                                break;
+                            case POST:
+                                headers.setMethod(HttpRequestHeader.POST);
+                                break;
+                            default:
+                                throw new IllegalArgumentException(
+                                        "Method " + method + " not supported.");
+                        }
+                        msg.setRequestHeader(headers);
+                        httpSender.sendAndReceive(msg);
+                        addCacheMessage(msg);
+                        try {
+                            Thread.sleep(2000);
+                        } catch (InterruptedException e) {
+                            // TODO Auto-generated catch block
```

Comment:
These should be removed, extract a static method for the sleep. (There's one case that the interrupted status is not being restored.)